src/api.v2/controllers/subsetController.js

```diff
@@ -10,66 +9,86 @@ const UserAccess = require('../model/UserAccess');
 
 const logger = getLogger('[SubsetController] - ');
 
-const runSubset = async (req, res) => {
-  const { experimentId } = req.params;
-
-  logger.log(`Starting subset for experiment ${experimentId}`);
-
+const runSubset = async (stateMachineParams, authorization) => {
   const {
-    params: { experimentId: parentExperimentId },
-    body: { name, cellSetKeys },
-    user: { sub: userId },
-  } = req;
+    experimentId: parentExperimentId,
+    name,
+    userId,
+  } = stateMachineParams;
 
-  logger.log(`Creating experiment to subset ${parentExperimentId}`);
+  // Doesn't exist initially
+  let { subsetExperimentId } = stateMachineParams;
 
-  let subsetExperimentId;
-  await sqlClient.get().transaction(async (trx) => {
-    subsetExperimentId = await new Experiment(trx).createCopy(parentExperimentId, name);
-    await new UserAccess(trx).createNewExperimentPermissions(userId, subsetExperimentId);
+  // Upon retrying the subset experiment id already exists,
+  // and we don't want to create a new experiment
```

Comment:
I didn't understand this comment, is this run when retrying?

Reply:
I separated this function into two: `createExperimentToSubset` and `executeSubsetPipeline`. The `/subset` endpoint first calls `createExperimentToSubset` and then `executeSubsetPipeline`, where as `/retry` just calls `executeSubsetPipeline`.

Still cannot evade the need of a wrapper function for `req` and `res` as I need to build the `params` object for `executeSubsetPipeline`. 

Still a pretty good improvement imo